js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx

```diff
@@ -96,18 +107,70 @@ export function buildAssetCountBySection(assets: AssetDefinitionMetadata[]): Ass
         (assetCountByCodeLocation[stringifiedCodeLocation] || 0) + 1;
     });
 
-  const countPerAssetGroup = Object.entries(assetCountByGroup).map(([groupIdentifier, count]) => ({
-    assetCount: count,
-    groupMetadata: JSON.parse(groupIdentifier),
-  }));
+  const countsByOwner = Object.entries(assetCountByOwner)
+    .map(([owner, count]) => ({
+      owner,
+      assetCount: count,
+    }))
+    .sort(({owner: ownerA}, {owner: ownerB}) => COMMON_COLLATOR.compare(ownerA, ownerB));
+  const countsByComputeKind = Object.entries(assetCountByComputeKind)
+    .map(([computeKind, count]) => ({
+      computeKind,
+      assetCount: count,
+    }))
+    .sort(({computeKind: computeKindA}, {computeKind: computeKindB}) =>
+      COMMON_COLLATOR.compare(computeKindA, computeKindB),
+    );
+  const countPerAssetGroup = Object.entries(assetCountByGroup)
+    .map(([groupIdentifier, count]) => ({
+      assetCount: count,
+      groupMetadata: JSON.parse(groupIdentifier),
+    }))
+    .sort(
+      (
+        {
+          groupMetadata: {
+            groupName: groupNameA,
+            repositoryLocationName: repositoryLocationNameA,
+            repositoryName: repositoryNameA,
+          },
+        },
+        {
+          groupMetadata: {
+            groupName: groupNameB,
+            repositoryLocationName: repositoryLocationNameB,
+            repositoryName: repositoryNameB,
+          },
+        },
```

Comment:
This all looks great (and I like the fallback comparison using `||`)

It's a bit unusual to destructure this much in the function declaration -- i wonder if this would be a bit more succinct if you only set `{groupMetadata: groupA}, {groupMetadata: groupB}` and then access everything as `groupA.groupName` below